src/dynamic_boundary_conditions/tide/main_tide_slr.py

```diff
@@ -26,6 +28,161 @@
+                                f"Must be one of {list(valid_parameters.keys())}.")
+
+    # Check if the provided SSP scenario is valid
+    valid_ssp_scenarios = valid_parameters[confidence_level]['ssp_scenarios']
+    if ssp_scenario not in valid_ssp_scenarios:
+        return ValidationResult(False,
+                                f"Invalid value '{ssp_scenario}' for ssp_scenario."
+                                f" Must be one of {valid_ssp_scenarios}.")
+
+    # Check if the provided add_vlm value is valid
+    valid_add_vlms = [True, False]
+    if add_vlm not in valid_add_vlms:
+        return ValidationResult(False,
+                                f"Invalid value '{add_vlm}' for add_vlm. Must be one of {valid_add_vlms}.")
+
+    # Get the percentiles from the column names and check if the provided percentile value is valid
+    column_names_query = text(r"""
+        SELECT column_name
+        FROM information_schema.columns
+        WHERE
+            table_name = 'sea_level_rise'
+            AND table_catalog=:db_name
+            AND column_name ~ '^p\d+'
+        """).bindparams(db_name=config.get_env_variable("POSTGRES_DB"))
+    percentile_col_tuples = engine.execute(column_names_query).fetchall()
+    # Flatten the result of the query
+    percentile_cols = [col_tuple[0] for col_tuple in percentile_col_tuples]
+    # Remove the leading 'p'
+    valid_percentiles = [int(col[1:]) for col in percentile_cols]
+
+    if percentile not in valid_percentiles:
+        return ValidationResult(
+            False,
+            f"Invalid value '{percentile}' for percentile. Must be one of {valid_percentiles}.")
+
+    # Check if the provided projection year is valid
+    min_year = valid_parameters[confidence_level]['min_year']
+    max_year = valid_parameters[confidence_level]['max_year']
+    valid_proj_year = range(min_year, max_year + increment_year, increment_year)
+    if proj_year not in valid_proj_year:
+        return ValidationResult(
+            False,
+            f"Invalid value '{proj_year}' for proj_year. Must be one of {list(valid_proj_year)}.")
+
+    # Pass the validation
+    return ValidationResult(True, None)
+
+
+def get_valid_parameters_based_on_confidence_level() -> Dict[str, Union[str, int]]:
+    """
+    Get information on valid tide and sea-level-rise parameters based on the valid values in the database.
+    These parameters are mostly dependent on the "confidence_level" parameter, so that is the key in the returned dict.
+
+    Returns
+    -------
+    Dict[str, Union[str, int]]
+        Dictionary with confidence_level as the key, and the allowed values for other items as values.
+    """
+    # Connect to database
+    engine = setup_environment.get_database()
+    slr_table_name = 'sea_level_rise'
+    if not tables.check_table_exists(engine, slr_table_name):
+        # Sea Level Rise data has not been initialised, so initialise it.
+        sea_level_rise_data.store_slr_data_to_db(engine)
```

Comment:
That is not the only use of this function, it is also used in the new endpoint. Without this front end can not run, so I needed to initalise.

Reply:
I noticed it’s used in `app.py` `tasks.py`, so I assume that with the new endpoint, it will need the sea level rise data for the front end to work before processing reaches the backend `main()` function?

Reply:
Yes, the function is called to retrieve parameters for the user, before the user requests a scenario